explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
+subscribe to push messages.
+
+### Periodic Background Sync
+
+[Periodic Background
+Sync](https://github.com/WICG/BackgroundSync/blob/master/explainer.md#periodic-synchronization-in-design)
+is a proposed extension to the [Background
+Sync](https://wicg.github.io/BackgroundSync/spec/) API, that allows a service
+worker to periodically poll the server, which could be used to get an updated
+status and call `Badge.set`. However, this API is unreliable: the period that it
+gets called is at the discretion of the user agent and can be subject to things
+like battery status.
+
+That means when the page isn't open, you could have the badge indicator update
+every once in awhile, but have no guarantee that it would be up to date.
+
+(Also, the Periodic Background Sync API is not yet implemented.)
+
+It's possible that the use of Push API and Periodic Background Sync together is
+"good enough": high-priority notices (that show a notification) come in through
+a Push and are displayed immediately with a notification; low-priority notices
+are displayed as badges immediately if there are pages open, or "eventually" if
+there are no pages open.
+
+However, in our view, there is a need for *immediate* update to low-priority
+notices: this is akin to the [urgency vs importance](https://en.wikipedia.org/wiki/Time_management#The_Eisenhower_Method)
+dichotomy: a badge is for non-important information but that doesn't mean the
+user doesn't want to see the notice in a timely fashion.
+
+### Possible changes to the Push API
+
+Here we discuss ways to potentially allow the Push API to update badges without
+showing notifications.
+
+**Note: This should not be considered blocking.** The Badge API is perfectly
+usable, including from service workers, without these changes, so we consider
+these as add-ons that we could introduce at a later time.
+
+#### Waiving the notification requirement
+
+The purpose of the [de facto
+requirement](https://github.com/w3c/push-api/issues/313) to show a notification
+upon receipt of a push message is to prevent malicious sites from using
+high-frequency pushes to do invisible background work, such as crypto mining
+(use of the user's computing resources) or monitoring the user's IP address,
+giving coarse-grained geographic location (privacy concern). The theory is that
+if the site is required to show a notification on each push, the user is at
+least going to be aware that the site is constantly doing something, and if it's
+unreasonably spammy, the user is likely to revoke the notification permission,
+thus erasing the push subscription.
+
+We could make it so that use of the `Badge` API serves the same function as
+showing a notification (fulfilling the requirement to use the Push API). But
+doing so would probably nullify the reason for this requirement in the first
+place: a malicious site that wants to do crypto mining could just set a badge
+every 30 seconds and the user would probably not notice it.
```

Comment:
They could set it to the same value, and the user **definitely** wouldn't notice it.

or

```js
const onNotificationReceived = () => {
    Badge.set();
    Badge.clear();
};
```

Reply:
Done.